ext4/dmverity/dmverity_test.go

```diff
@@ -0,0 +1,102 @@
+package dmverity
+
+import (
+	"bytes"
+	"encoding/binary"
+	"io"
+	"io/ioutil"
+	"math/rand"
+	"os"
+	"strings"
+	"testing"
+	"unsafe"
+
+	"github.com/pkg/errors"
+)
+
+func tempFileWithContentLength(t *testing.T, length int) *os.File {
+	tmpFile, err := ioutil.TempFile("", "")
+	if err != nil {
+		t.Fatalf("failed to create temp file")
+	}
+	defer tmpFile.Close()
+
+	content := make([]byte, length)
+	if _, err := rand.Read(content); err != nil {
+		t.Fatalf("failed to write random bytes to buffer")
+	}
+	if _, err := tmpFile.Write(content); err != nil {
+		t.Fatalf("failed to write random bytes to temp file")
+	}
+	return tmpFile
+}
+
+func writeDMVeritySuperBlock(filename string) (*os.File, error) {
+	out, err := os.OpenFile(filename, os.O_RDWR, 0777)
+	if err != nil {
+		return nil, err
+	}
+	defer out.Close()
+	fsSize, err := out.Seek(0, io.SeekEnd)
+	if err != nil {
+		return nil, err
+	}
+	sb := NewDMVeritySuperblock(uint64(fsSize))
+	if err := binary.Write(out, binary.LittleEndian, sb); err != nil {
+		return nil, err
+	}
+	sbSize := int(unsafe.Sizeof(*sb))
+	padding := bytes.Repeat([]byte{0}, blockSize-(sbSize%blockSize))
+	if _, err = out.Write(padding); err != nil {
+		return nil, err
+	}
+	return out, nil
+}
+
+func TestInvalidReadEOF(t *testing.T) {
+	tmpFile := tempFileWithContentLength(t, blockSize)
+	_, err := ReadDMVerityInfo(tmpFile.Name(), blockSize)
+	if err == nil {
+		t.Fatalf("no error returned")
+	}
+	if errors.Cause(err) != ErrSuperBlockReadFailure || !strings.Contains(err.Error(), "EOF") {
```

Comment:
For the second half, we can't check if it's `io.EOF`?

Reply:
Or rather, not `io.EOF`

Reply:
portion of the code this unit test is targeting for context:
```
		if err != nil {
			return nil, errors.Wrapf(ErrSuperBlockReadFailure, "%s", err)
		}
```
unfortunately checking against `io.EOF` won't work, since we're wrapping. We could, instead just do ` return nil, fmt.Errorf("failed to read super-block: %w", err)` and check for `errors.Unwrap(err) == io.EOF` in the unit test, which would probably make more sense, but not the main objective of this PR.

Reply:
I actually changed the error wrapping. I think it wasn't right before.